### PR TITLE
[Waiting for 3.0] Remove deprecated APIs in table node protocols

### PR DESCRIPTION
**Pull request by nguyenhuy**
Those APIs were deprecated late last year (pre-2.0).

### CODE REVIEW

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Any source code changes should have an entry in CHANGELOG.md or have #trivial in their title.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/881d3805eb7b365eacd81bf8d57f11bd/raw/878d516acef60bac56b47828d88fd28ccaf4ff43/log.txt)

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This is a big PR, please consider splitting it up to ease code review.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: Tests/ASTableViewTests.mm [hunk 1]
```diff
@@ -270,31 +268,31 @@ - (void)testConstrainedSizeForRowAtIndexPath
...
-  tableView.asyncDataSource = delegate;
-  tableView.asyncDelegate = delegate;
-
-//  [delegate release];
-  XCTAssertTrue(delegateDidDealloc, @"unexpected delegate lifetime:%@", delegate);
-  
-//  XCTAssertNoThrow([tableView release], @"unexpected exception when deallocating table view:%@", tableView);
-  XCTAssertTrue(tableViewDidDealloc, @"unexpected table view lifetime:%@", tableView);
-}
+//- (void)DISABLED_testTableViewDoesNotRetainItselfAndDelegate
```
Comment (garrettmoon) [inline Tests/ASTableViewTests.mm]: Just curious here, why comment these out?

Comment (nguyenhuy) [reply Tests/ASTableViewTests.mm]: I'm lazy to convert them to the right APIs.

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This is a big PR, please consider splitting it up to ease code review.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This is a big PR, please consider splitting it up to ease code review.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by nguyenhuy** [comment]
This PR is too old now. I’m gonna close it and open a new one when the time comes.

**Pull request closed without merging**